lib/utils.py

```diff
@@ -126,3 +128,16 @@ async def send_with_gzip(self, message: Message) -> None:
 
 async def unattached_send(message: Message) -> typing.NoReturn:
     raise RuntimeError("send awaitable not set")  # pragma: no cover
+
+
+def to_python_primitive(v):
+    if hasattr(v, "source"):
```

Comment:
It will return a str() version of it.

I could check instance of Function here, but then I would have to check if it issubclass of Function as well because there could be custom Functions in the future.

I prefer to keep it very simple as this is a temporary approach. Ideally we will refactor that in the future to use official encoders instead.